static/js/src/public/details/resources/index.js

```diff
@@ -0,0 +1,63 @@
+function handleTogglePanels() {
+  const panelToggleButtons = document.querySelectorAll(".js-panel-toggle");
+
+  panelToggleButtons.forEach((panelToggleButton) => {
+    panelToggleButton.addEventListener("click", (e) => {
+      const target = e.target;
+      const panel = document.getElementById(
+        target.getAttribute("aria-controls")
+      );
+
+      target.classList.toggle("is-active");
+      panel.classList.toggle("is-open");
+
+      if (panel.classList.contains("is-open")) {
+        panel.querySelector("[readonly]").focus();
```

Comment:
Last thing, and then it's good to go - could you toggle a `role="dialog"` attribute on the `panel`? At the moment, a screen reader doesn't announce anything when the dropdown appears, but it does if that attribute is added when it's shown. It reads out the entire key plus the word "selected", which is a mouthful, but I think that's better than nothing.